### PR TITLE
refactor: Remove `autodiff`

### LINKED ISSUE #2792: Remove autodiff documentation
https://acts.readthedocs.io/en/latest/plugins/autodiff.html#id1

### PR DESCRIPTION
**Pull request by andiwand**
related issues (closes #2792)
- https://github.com/acts-project/acts/issues/2792

`autodiff` became more of a burden at some point. Since we do not consistently template all the quantities we might want to derive later this became more of an exception. I propose to remove it for now until we have a broader solution or purpose for it

### CODE REVIEW

**Review by asalzburger — DISMISSED** [review]
Clean removal, as discussed, we should do that.

**Comment by asalzburger** [comment]
@andiwand - can you just resolve the conflict, please?

**Comment by paulgessinger** [comment]
I would like to hold this until @benjaminhuth has had a chance to look at this.

**Comment by benjaminhuth** [comment]
Hmm so I'm fine with removing the plugin. However, I used the template infrastructure as well for my experimental SIMD stepper implementation for the GSF which I did not try to merge yet...

Not sure if that's an argument to keep some of the template infrastructure (mainly that of `detail::GenericDefaultExtension`), and I try to merge it as `Acts::Experimental`?

However since this is still very experimental you could also remove all of this and I would add it again in case that becomes relevant in the future...

**Comment by andiwand** [comment]
I wound suggest to add the templating back in if necessary. Happy to help when this comes up

File: Core/include/Acts/Propagator/DefaultExtension.hpp [hunk 1]
```diff
@@ -1,18 +1,245 @@
...
+
+#include <array>
 
 namespace Acts {
 
-/// @brief A typedef for the default GenericDefaultExtension with double.
-using DefaultExtension = detail::GenericDefaultExtension<double>;
+/// @brief Default evaluater of the k_i's and elements of the transport matrix
+/// D of the RKN4 stepping. This is a pure implementation by textbook.
+struct DefaultExtension {
```
Comment (paulgessinger) [inline Core/include/Acts/Propagator/DefaultExtension.hpp:21]: This just *inlines* the `GenericDefaultExtension` with `double` as template argument, yes?

**Review by paulgessinger — APPROVED** [review]

File: Core/include/Acts/Propagator/DefaultExtension.hpp (hunk 1 above)
Comment (andiwand) [reply Core/include/Acts/Propagator/DefaultExtension.hpp:21]: yes exactly

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [c2e1e6e748cefb13c2eb9408894b5831e757805a]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868487)||
|:yellow_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868486)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868485)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868484)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868483)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868482)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868481)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868480)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868479)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868478)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868477)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868476)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868475)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868474)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868473)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868472)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868471)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868470)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868469)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868468)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868467)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868466)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868465)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35868464)||

**Pull request merged**